flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/TemporalTypesTest.scala

```diff
@@ -795,8 +794,8 @@ class TemporalTypesTest extends ExpressionTestBase {
   def testTemporalShanghai(): Unit = {
     config.setLocalTimeZone(ZoneId.of("Asia/Shanghai"))
 
-    testSqlApi(timestampTz("2018-03-14 19:01:02.123"), "2018-03-14 19:01:02.123")
-    testSqlApi(timestampTz("2018-03-14 19:00:00.010"), "2018-03-14 19:00:00.01")
+    testSqlApi(timestampTz("2018-03-14 19:01:02.123"), "2018-03-14 19:01:02.123000")
```

Comment:
Currently, we don't have a way to construct TIMESTAMP WITH LOCAL TIME ZONE literal. `timestampTz` will lose the original timestamp literal precision and use a default 6 precision. The tests are counter-intuitive because we are padding zeros for a timestamp literal. 

Could we improve the `timestampTz` a little bit? It can extract precision from the literal string, and cast to the local timestamp with the extracted precison.

Reply:
`timestampTz(expr)` is a shortcut of `CAST(expr AS TIMESTAMP WITH LOCAL TIME ZONE)`, Do you mean we should provide a builtin UDF to convert TIMESTAMP literal to TIMESTAMP WITH LOCAL TIME ZONE?

Reply:
I mean we should improve `timestampTz` to convert timestamp without precision loss. 

```scala
private def timestampTz(str: String) = {
  val precision = extractPrecision(str)
   s"CAST(TIMESTAMP '$str' AS TIMESTAMP($precision) WITH LOCAL TIME ZONE)"
}
```

Reply:
OK, According to SQL standard,  for timestamp literal, the length of the second fraction should be the precision.